examples/fungible/src/state.rs

```diff
@@ -63,9 +68,16 @@ impl FungibleToken {
         balance
             .try_sub_assign(amount)
             .map_err(|_| InsufficientBalanceError)?;
-        self.accounts
-            .insert(&account, balance)
-            .expect("Failed insertion operation");
+        if balance == Amount::ZERO {
+            match self.accounts.remove(&account) {
+                Ok(()) | Err(ViewError::NotFound(_)) => {}
```

Comment:
Why does the `ViewError::NotFound` not result in an error?

Reply:
I thought I'd handle the case gracefully where you debit `0` from an empty account. In that case we would call `remove`, but the account is already not there…

But you're right, I should just check whether the debited amount is `0` right away, and then just return. I changed it.